Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +13,70 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
+        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            
+             TimeSpan max = TimeSpan.Parse("0");
+            for (int i = 0; i < exactPostTime.Length; i++)
+            {
+                for (int j = i + 1; j < exactPostTime.Length; j++)
+                {
+                    if (exactPostTime[i] == exactPostTime[j])
+                        if (showPostTime[i] != showPostTime[j])
+                            return "impossible";
+                }
+            }
+            for (int i = 0; i < exactPostTime.Length; i++)
+            {
+                TimeSpan time = TimeSpan.Parse(exactPostTime[i]);
+                if (showPostTime[i].Contains("sec"))
+                {
+                    if (max < time)
+                        max = time;
+
+                }
+                else
+                {
+                    int val = int.Parse(showPostTime[i].Substring(0, showPostTime[i].IndexOf(" ")));
+                    if (showPostTime[i].Contains("min"))
+                    {
+                        TimeSpan span = TimeSpan.FromMinutes(val);
+                        if (time + span > TimeSpan.Parse("1.00:00:00"))
+                        {
+                            time = time + span - TimeSpan.FromDays(1);

```

Comment:
there is repetitive code inside if conditions which can be moved into private method to avoid repetition